Finalize patch documentation + Fix nested do block bug

EthanLuisMcDonough opened the pull request:
The first two commits mostly focus on adding additional documentation, especially in einfo.ads and sinfo.ads. The last commit fixes an issue that occurs when an empty exit appears inside a `parallel do` construct that is nested inside a loop.

sttaft commented on gcc/ada/sinfo.ads:
```diff
@@ -1506,6 +1510,10 @@ package Sinfo is
 
+   --  In_Outlined_Parallel
+   --    Set when a N_Parallel_Block_Statement or N_Loop_Statement node
+   --    has been relocated into an outlined parallel procedure.
```
Perhaps just say "procedure" here rather than "parallel procedure".  There is nothing "parallel" about the procedure itself.

sttaft commented on gcc/ada/sinfo.ads:
```diff
@@ -2171,6 +2189,29 @@ package Sinfo is
    --    N_Others_Choice entry.
 
+   --  Parallel_Chunk_Id
+   --    Applies to N_Loop_Statements with parallel iteraton schemes. Stores
```
"iteraton" => "iteration"

sttaft commented on gcc/ada/sinfo.ads:
```diff
@@ -2171,6 +2189,29 @@ package Sinfo is
+   --    iteration schemes. Neither of these statements have room for
+   --    declarations in their syntax, but they are later moved inside of an
+   --    outlined procedure before analysis. This list is used to collect block
+   --    and loop labels that are created inside Par.Labl so that they can be
```
What is "Par.Labl"?  Seems like it might be referring to some specific code, whereas this paragraph should probably be at a somewhat higher level.

sttaft commented on gcc/ada/sem_ch6.adb:
```diff
@@ -3765,7 +3821,9 @@ package body Sem_Ch6 is
                --  Rewrite exits that exit a loop outside the current
-               --  parallel loop
+               --  parallel loop. We make sure not to include the condition
+               --  in the 
```
"in the" ... ?

sttaft commented on gcc/ada/sem_ch6.adb:
```diff
@@ -3721,13 +3748,39 @@ package body Sem_Ch6 is
+
             if Nkind (I) = N_Simple_Return_Statement then
+               --  For return statements that return a value,
+               --  move the return value the Return_Val variable
```
"move the return value *into* the Return_Val variable"

sttaft reviewed (changes requested):
Just a few typographical errors.

EthanLuisMcDonough commented:
Okay, I fixed the documentation issues. I also added some additional checks to the chunk index pre-analysis processing. I noticed that I was getting internal compiler errors when the chunk specifier discrete subtype definition wasn't a discrete range type.

reznikmm commented on gcc/ada/sem_ch5.adb:
```diff
@@ -4799,37 +4839,42 @@ package body Sem_Ch5 is
          Analyze_Chunk_Specification (Chunk_Spec);
       end if;
 
+      --  Pre-analysis transformations for parllel expansion.
```
```suggestion
      --  Pre-analysis transformations for parallel expansion.
```
